src/commonMain/kotlin/fr/acinq/lightning/channel/Commitments.kt

```diff
@@ -421,7 +422,18 @@ data class Commitment(
             "built remote commit number=${remoteCommit.index + 1} toLocalMsat=${spec.toLocal.toLong()} toRemoteMsat=${spec.toRemote.toLong()} htlc_in=$htlcsIn htlc_out=$htlcsOut feeratePerKw=${spec.feerate} txId=${remoteCommitTx.tx.txid} fundingTxId=$fundingTxId"
         }
 
-        val commitSig = CommitSig(params.channelId, sig, htlcSigs.toList())
+        val tlvs = buildSet {
+            if (spec.htlcs.isEmpty()) {
+                val alternativeSigs = Commitments.alternativeFeerates.map { feerate ->
+                    val alternativeSpec = spec.copy(feerate = feerate)
+                    val (alternativeRemoteCommitTx, _) = Commitments.makeRemoteTxs(channelKeys, commitTxNumber = remoteCommit.index + 1, params.localParams, params.remoteParams, fundingTxIndex = fundingTxIndex, remoteFundingPubKey = remoteFundingPubkey, commitInput, remotePerCommitmentPoint = remoteNextPerCommitmentPoint, alternativeSpec)
+                    val alternativeSig = Transactions.sign(alternativeRemoteCommitTx, channelKeys.fundingKey(fundingTxIndex))
+                    CommitSigTlv.AlternativeFeerateSig(feerate, alternativeSig)
+                }
+                add(CommitSigTlv.AlternativeFeerateSigs(alternativeSigs))
+            }
```

Comment:
This code should be called by InteractiveTx.create(), otherwise we are not providing additional sigs at channel creation and for splice txs.

Reply:
Good catch, I'll add that.

Reply:
Done in https://github.com/ACINQ/lightning-kmp/pull/553/commits/8032630c6533153d7f226be5069b5bbebb4111d4 with additional tests.